frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -2046,6 +2050,24 @@ func validateStageNames(stages []instructions.Stage, warn linter.LintWarnFunc) {
 	}
 }
 
+func reportUnmatchedVariables(cmd instructions.Command, buildArgs []instructions.KeyValuePairOptional, unmatched map[string]struct{}, opt *dispatchOpt) {
+	for unmatch := range unmatched {
+		for _, buildArg := range buildArgs {
```

Comment:
Double `for` is not needed here. Just

```
for _, buildArg := range buildArgs {
   delete(unmatched, buildArg.Key)
}
```

You can also add early return if `len(unmatched) == 0`

Reply:
Excellent! Fixed!